crypto/evp/kdf_lib.c

```diff
@@ -70,66 +104,56 @@ void EVP_KDF_reset(EVP_KDF_CTX *ctx)
         return;
 
     if (ctx->meth->reset != NULL)
-        ctx->meth->reset(ctx->impl);
+        ctx->meth->reset(ctx->data);
 }
 
-int EVP_KDF_ctrl(EVP_KDF_CTX *ctx, int cmd, ...)
+size_t EVP_KDF_size(EVP_KDF_CTX *ctx)
 {
-    int ret;
-    va_list args;
-
-    va_start(args, cmd);
-    ret = EVP_KDF_vctrl(ctx, cmd, args);
-    va_end(args);
-
-    if (ret == -2)
-        EVPerr(EVP_F_EVP_KDF_CTRL, EVP_R_COMMAND_NOT_SUPPORTED);
+    OSSL_PARAM params[2] = { OSSL_PARAM_END, OSSL_PARAM_END };
+    size_t s;
 
-    return ret;
-}
-
-int EVP_KDF_vctrl(EVP_KDF_CTX *ctx, int cmd, va_list args)
-{
     if (ctx == NULL)
         return 0;
 
-    return ctx->meth->ctrl(ctx->impl, cmd, args);
+    if (ctx->meth->get_ctx_params != NULL) {
+        *params = OSSL_PARAM_construct_size_t(OSSL_KDF_PARAM_SIZE, &s);
+        if (ctx->meth->get_ctx_params(ctx, params))
+            return s;
+    }
+    return SIZE_MAX;
```

Comment:
I wonder if this is appropriate...  I think I'd rather return zero here, and have each implementation specify a size, where `SIZE_MAX` is absolutely fine.  That could serve as an indication that something isn't quite right, i.e. the caller would know that there's an error record in the error queue

Reply:
Okay, I think this is what the previous code path returned -- PRFs can return any amount of output.  I'm fine with implementing the size call for all, so I think the change makes sense.

Reply:
I don't think there was such an original.  Looking at 1.1.1, the closest I can find is this (since KDFs were implemented as EVP_PKEY methods):

``` C
int EVP_PKEY_size(const EVP_PKEY *pkey)
{
    if (pkey && pkey->ameth && pkey->ameth->pkey_size)
        return pkey->ameth->pkey_size(pkey);
    return 0;
}
```

So I believe `SIZE_MAX` was utterly and completely a new value for the new KDF API, which hasn't been released yet, so we're free to redefine it as much as we want.

Reply:
Changed to return zero.